### ISSUE TITLE
-O flag does not work with luaCsnd6

### ISSUE THREAD

**Issue by levinericzimmermann** [csound/csound#1994]
Hello everyone,

thanks for maintaining and working on Csound and keeping the project alive!

I'm currently [working on a small project](https://codeberg.org/wm2/wm2) using Csounds Lua API. I would like to have logfiles with Csound log outputs, however this currently doesn't seem to be possible with the `-O` option. I added a [minimal test here](https://codeberg.org/wm2/wm2/src/branch/test-cs-logfile/cs-log-test.lua). I guess this is related to https://github.com/csound/csound/issues/375. I know @jpffitch said in https://github.com/csound/csound/issues/375, this is "not possible in any reasonable way", however this is already 10 years ago and I'm wondering if things have changed now?

I realized, when using the Lua API the Csound log output is send to stderr. This makes it possible [to manually create csound logfiles](https://codeberg.org/wm2/wm2/src/branch/test-cs-logfile/test#L3), but this is a bit clumsy (and hides real stderr of the program), so I would prefer if the `-O` option would work.

I tested with Csound 6.18.1,  NixOS 24.05 (Linux).

Any help is very much appreciated!

Best, Levin

**Comment by vlazzarini** [comment]
I'm afraid you might need to implement this by hand, I guess taking the message output via a callback. The flag is implemented in the CLI frontend, not in CsoundLib, so the API does not have access to it.

Furthermore, Csound 6.x is EOL, the swig wrappers are not supported in Csound 7.

Also afaik the original author of the lua swig wrapper stopped actively maintaining it a while ago.

**Comment by levinericzimmermann** [comment]
Thanks for your feedback, Victor, this is quite helpful. It's sad that the Lua SWIG wrapper isn't maintained anymore. I'm considering to switch to [`luajit FFI`](https://luajit.org/ext_ffi.html).

I tried a quick proof-of-concept and [replicated my previous minimal example with it](https://codeberg.org/wm2/wm2/src/commit/ebc2d121946fdf5076c2f28aa934c75231953151/cs-ffi-log-test.lua) which worked fine so far. I followed your suggestion and set the callback in my script which also worked fine ([1](https://codeberg.org/wm2/wm2/src/commit/ebc2d121946fdf5076c2f28aa934c75231953151/t.c#L34-L37), [2](https://codeberg.org/wm2/wm2/src/commit/ebc2d121946fdf5076c2f28aa934c75231953151/cs-ffi-log-test.lua#L26-L28)): I can use Csound with logfiles now.

Earlier or later, i write a proper `csound` module with `luajit FFI` to switch my project to this. It shouldn't take very long. If it becomes stable, it may be useful for anyone else wanting to use Csound from Lua.

Best, Levin
